### PR TITLE
Fix property form breaking when not all fields are present

### PR DESCRIPTION
**Pull request by joeybrk372**
- [x] @sfeng88 @rygim @jharwig 
- [x] @srfarley @joeferner
- [x] @EvanOxfeld @joeybrk372
- [x] @mwizeman @dsingley

### CODE REVIEW

File: web/war/src/main/webapp/js/detail/dropdowns/propertyForm/propForm.js [hunk 1]
```diff
@@ -470,25 +475,25 @@ define([
...
+                    (this.justification ? this.justification.valid : true);
                 var empty = _.reject(this.$node.find('.configuration input'), function(input) {
                     return !!input.value;
                 }).length > 0;
 
                 this.valid = valid && !empty;
             }
 
-            if (this.valid && this.attr.modified) {
+            if (this.valid && _.contains(_.values(this.modified), true)) {
```
Comment (jharwig) [inline web/war/src/main/webapp/js/detail/dropdowns/propertyForm/propForm.js]: Can be simplified to `_.some(this.modified)`

File: web/war/src/main/webapp/js/detail/dropdowns/propertyForm/propForm.js [hunk 2]
```diff
@@ -530,6 +549,12 @@ define([
             _.defer(this.clearLoading.bind(this));
         };
 
+        this.getConfigurationValues = function() {
+            var instanceInfo = flightRegistry.findInstanceInfoByNode(this.select('configurationSelector').get(0)).shift(),
```
Comment (jharwig) [inline web/war/src/main/webapp/js/detail/dropdowns/propertyForm/propForm.js]: We have a jquery plugin for this. `jquery.flight.js`

`this.select('configurationSelector').lookupAllComponents()`

File: web/war/src/main/webapp/js/detail/dropdowns/propertyForm/propForm.js [hunk 3]
```diff
@@ -470,25 +475,25 @@ define([
             } else {
                 var valid = !this.propertyInvalid &&
                     (this.visibilitySource && this.visibilitySource.valid) &&
-                    (this.justification && this.justification.valid);
+                    (this.justification ? this.justification.valid : true);
```
Comment (EvanOxfeld) [inline web/war/src/main/webapp/js/detail/dropdowns/propertyForm/propForm.js]: I prefer `this.justification && this.justification.valid || !this.justification`. Ternary statements are hard to read when they're nested or within a long set of conditionals.

Comment (srfarley) [inline web/war/src/main/webapp/js/detail/dropdowns/propertyForm/propForm.js]: @EvanOxfeld I find the ternary expression much easier to read. I had to think longer to understand your version. That's my opinion. Sorry.

Comment (EvanOxfeld) [inline web/war/src/main/webapp/js/detail/dropdowns/propertyForm/propForm.js]: @srfarley FWIW I still gave the pull request a 👍 .

Comment (joeybrk372) [inline web/war/src/main/webapp/js/detail/dropdowns/propertyForm/propForm.js]: I find the ternary easier to read here as well.

**Pull request merged**